aws_google_auth/prepare.py

```diff
@@ -17,10 +17,11 @@ def get_prepared_config(
     def default_if_none(value, default):
         return value if value is not None else default
 
-    google_config.profile = default_if_none(profile, google_config.profile)
-
+    google_config.profile = profile
```

Comment:
I think this should be left as it is, and change the default from 
`config.profile = session.profile or 'default'` to `config.profile = session.profile or None`

I think that will allow a stored profile to be retained, but a new profile to be considered as `None`.
I have not validated that session.profile is None when not set at the CLI, so YMMV.